### ISSUE TITLE
Rerunning tailscale up command with --snat-subnet-routes=false does not remove SNAT rule

### ISSUE THREAD

**Issue by mprasil** [tailscale/tailscale#8007]
I've noticed that when I have already active connection and re-run the tailscale up adding `--snat-subnet-routes=false`, command runs fine, but SNAT rules aren't dropped.

Doing `tailscale down && tailscale up` afterwards drops the SNAT rule as expected.

This feels like a bug, but if it's intended behavior, it should be documented.

**Comment by DentonGentry** [comment]
Removal of the rule is supposed to happen here: https://github.com/tailscale/tailscale/blob/1f4a34588be4b0da4838667f11fe4fef6c285d1f/wgengine/router/router_linux.go#L1541

A quick test here does appear to have removed the iptables rule.

Could I ask you to repeat it, with a bug report?
1. `tailscale up --snat-subnet-routes=true`
1. `tailscale up --snat-subnet-routes=false`
1. `tailscale bugreport`

Something appears to have gone wrong in trying to remove the masquerade rule.

**Comment by mprasil** [comment]
Thanks for reply @DentonGentry, I tried today, but can't reproduce it anymore. Apologies, I should have tried to reproduce it before reporting bug. I must have hit some weird edge case maybe?

I'm going to close this now assuming I messed up something. I tried to search logs from around that time and so far found nothing suspicious, if I stumble upon something or if this reoccurs, I'll reopen. If you have any hints what to look for in the logs, let me know and I'll take a look.

Thanks for quick response.
